app/src/main/java/ru/netology/fmhandroid/adapter/NewsControlPanelAdapter.kt

```diff
@@ -0,0 +1,121 @@
+{
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NewsControlPanelViewHolder {
+        val binding = ItemNewsControlPanelBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+
+        return NewsControlPanelViewHolder(binding, onInteractionListener)
+    }
+
+    override fun onBindViewHolder(holder: NewsControlPanelViewHolder, position: Int) {
+        getItem(position)?.let {
+            holder.bind(it)
+        }
+    }
+
+    class NewsControlPanelViewHolder(
+        private val binding: ItemNewsControlPanelBinding,
+        private val onInteractionListener: NewsOnInteractionListener
+    ) : RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(newsItemWithCreator: NewsWithCreators) = binding.apply {
+            newsItemTitleTextView.text = newsItemWithCreator.news.newsItem.title
+            newsItemDescriptionTextView.text = newsItemWithCreator.news.newsItem.description
+            newsItemPublicationDateTextView.text =
+                newsItemWithCreator.news.newsItem.publishDate?.let { Utils.showDate(it) }
+            newsItemCreateDateTextView.text =
+                newsItemWithCreator.news.newsItem.createDate?.let { Utils.showDate(it) }
+            newsItemAuthorNameTextView.text = itemView.resources.getString(
+                R.string.full_name_format,
+                newsItemWithCreator.user.lastName,
+                newsItemWithCreator.user.firstName?.first()?.plus("."),
+                newsItemWithCreator.user.middleName?.first()?.plus(".")
+            )
+
+            setCategoryIcon(newsItemWithCreator)
+
+            newsItemMaterialCardView.setOnClickListener {
+                when (newsItemDescriptionTextView.visibility) {
+                    View.GONE -> {
+                        newsItemDescriptionTextView.visibility = View.VISIBLE
+                        viewNewsItemImageView.setImageResource(R.drawable.expand_less_24)
+                    }
+                    else -> {
+                        newsItemDescriptionTextView.visibility = View.GONE
+                        viewNewsItemImageView.setImageResource(R.drawable.expand_more_24)
+                    }
+                }
+            }
+
+            when (newsItemWithCreator.news.newsItem.publishEnabled) {
+                true -> {
+                    newsItemPublishedTextView.text =
+                        itemView.context.getString(R.string.news_control_panel_active)
+                    newsItemPublishedIconImageView.setImageResource(R.drawable.ic_baseline_check_24)
+                }
+                false -> {
+                    newsItemPublishedTextView.text =
+                        itemView.context.getString(R.string.news_control_panel_not_active)
+                    newsItemPublishedIconImageView.setImageResource(R.drawable.ic_baseline_clear_24)
+                }
+            }

```

Comment:
Серьёзно? `if ... else` похоже теперь стал "для слабаков" :smile: 
```suggestion
            if (newsItemWithCreator.news.newsItem.publishEnabled) {
                newsItemPublishedTextView.text = itemView.context.getString(R.string.news_control_panel_active)
                newsItemPublishedIconImageView.setImageResource(R.drawable.ic_baseline_check_24)
            } else {
                newsItemPublishedTextView.text = itemView.context.getString(R.string.news_control_panel_not_active)
                newsItemPublishedIconImageView.setImageResource(R.drawable.ic_baseline_clear_24)
            }
```